### PR TITLE
issue #401 - First round of changes to improve clarity of document. Removed mention of U-mode interrupts.

### PR DESCRIPTION
**Pull request by james-ball-qualcomm**
This PR contains my many updates to the CLIC spec to improve clarity. Based on discussions in the Fast Interrupt TG, I removed references to user-mode interrupts to simplify the specification. If the N extension someday moves forward towards ratification, we can add in support to the CLIC specification at that time.

### CODE REVIEW

File: src/clic.adoc [hunk 1]
```diff
@@ -428,124 +429,170 @@ be altered using `clicintattr[__i__]` and
...
+[%autowidth]
+|===
+| Extension Name | Description
+
+| smclic | CLIC support for M-mode
+| ssclic | CLIC support for S-mode
+| smclicshv | Selective Hardware Vectoring for M-mode
+| smclicconfig | Allows implementations to support different parameterizations of CLIC extensions
+|
+|===
```
Comment (christian-herber-nxp) [inline src/clic.adoc:447]: this table is giving an error in the build process

File: src/clic.adoc [hunk 2]
```diff
@@ -428,124 +429,170 @@ be altered using `clicintattr[__i__]` and
...
+
 
 == M-mode CLIC Register Access via Indirect CSR Access
 
-Access to CLIC registers clicinttrig[i], clicintip[i], clicintie[i], clicintattr[i], clicintctl[i] 
-is specified using the Indirect CSR Access extension method (Smcsrind/Sscsrind).  A CSR accessible via
-indirect CSR access may or may not be accessible via another method such as memory-mapped access.
+Access to CLIC registers clicintctl[i], clicintattr[i], clicintip[i], clicintie[i], and clicinttrig[i]
+utilizes the Indirect CSR Access extension (Smcsrind/Sscsrind). Implementations may support
```
Comment (christian-herber-nxp) [inline src/clic.adoc:453]: good i like it.

File: src/clic.adoc [hunk 3]
```diff
@@ -428,124 +429,170 @@ be altered using `clicintattr[__i__]` and
...
 
-----
- miselect  |  mireg |  mireg                    |  mireg2 |  mireg2                    |    description                   |
-           |  bits  |  register state           |  bits   |  register state            |                                  |
-0x1000+i   |  7:0   | RW  clicintctl[i x 4]     |   7:0   | RW  clicintattr[i x 4]     |  setting for interrupt i x 4     | 
-0x1000+i   | 15:8   | RW  clicintctl[i x 4 + 1] |  15:8   | RW  clicintattr[i x 4 + 1] |  setting for interrupt i x 4 + 1 |    
-0x1000+i   | 23:16  | RW  clicintctl[i x 4 + 2] |  23:16  | RW  clicintattr[i x 4 + 2] |  setting for interrupt i x 4 + 2 |    
-0x1000+i   | 31:24  | RW  clicintctl[i x 4 + 3] |  31:24  | RW  clicintattr[i x 4 + 3] |  setting for interrupt i x 4 + 3 |    
-----
+All CLIC registers are visible to M-mode.
```
Comment (christian-herber-nxp) [inline src/clic.adoc:461]: shouldn't this be covered by the privileged spec?

File: src/clic.adoc [hunk 4]
```diff
@@ -428,124 +429,170 @@ be altered using `clicintattr[__i__]` and
...
 
-In this miselect offset range, 
-Each mireg register controls the clic level/priority setting of four interrupts
-Each mireg2 register controls the clic attribute setting of four interrupts
+NOTE: Since accessing `clicintip[__i__]`, `clicintie[__i__]`, `clicintattr[__i__]`,
```
Comment (christian-herber-nxp) [inline src/clic.adoc:463]: not sure this note is needed.

File: src/clic.adoc [hunk 5]
```diff
@@ -657,24 +707,21 @@ positive level-triggered, negative level-triggered, positive edge-triggered,
...
 ----
- Encoding for RISC-V privilege levels (mstatus.mpp)
+ Encoding for RISC-V privilege modes (mstatus.mpp)
 
- Level  Encoding Name              Abbreviation
- 0      00       User/Application  U
- 1      01       Supervisor        S
- 2      10       Reserved
- 3      11       Machine           M
+ Mode Value   Name              Abbreviation
```
Comment (christian-herber-nxp) [inline src/clic.adoc:716]: why actually repeat this here at all?

File: src/clic.adoc [hunk 6]
```diff
@@ -912,104 +960,101 @@ through the trap handling process.
...
-{pp} and {pie}, appear as zero in the {cause} CSR but the corresponding
-state bits in the `mstatus` register are not cleared.
-
-Note: For now all privilege modes must run in either CLIC mode or all privilege modes must run in non-CLIC mode so switching to CLINT mode from CLIC mode causes {pil} in all privilege modes to be zeroed.
+Note: For now all privilege modes must run in either CLIC mode or all 
+privilege modes must run in non-CLIC mode so switching to CLINT mode 
+from CLIC mode causes {pil} in all privilege modes to be zeroed.
 
-when not in CLIC mode, {cause} has the CLINT mode format.
+When not in CLIC mode, {cause} has the CLINT mode format.
```
Comment (christian-herber-nxp) [inline src/clic.adoc:1003]: should be xcause?

**Comment by christian-herber-nxp** [comment]
i went roughly half through this PR for now. I gave some comments, but I don't think I can review this as a whole.
In general, it is good to limit the scope of a PR to one issue to enable a good review.
E.g. a single PR to fix the notion of mode/level would be easy to review.

**Comment by james-ball-qualcomm** [comment]
I’ll look at the table build error. Works fine in my VSCode preview but I should be building this first before doing a PR. Sorry, I’m still ramping up on these sorts of tasks.

**Review by christian-herber-nxp — APPROVED** [review]

File: src/clic.adoc (hunk 6 above)
Comment (james-ball-qualcomm) [reply src/clic.adoc:1002]: No, the ASCIIDOC converts {cause} to **x**cause automatically. BTW, this happens everywhere and for other CSR names too.

**Comment by james-ball-qualcomm** [comment]
I did the merge since it was approved (hope I'm supposed to do this, I'm still learning GitHub)

**Comment by christian-herber-nxp** [comment]
> I did the merge since it was approved (hope I'm supposed to do this, I'm still learning GitHub)

looks good. I would recommend to do a squash merge in future, to make to have one commit on the main branch per PR.

**Pull request merged**